x/dsmr/node.go

```diff
@@ -50,6 +49,12 @@ var (
 	ErrFailedToReplicate                   = errors.New("failed to replicate to sufficient stake")
 )
 
+type ChainState interface {
+	GetNetworkParams() (networdID uint32, subnetID, chainID ids.ID)
+	GetSignatureParams(ctx context.Context) (validators warp.CanonicalValidatorSet, quorumNum uint64, quorumDen uint64, err error)
```

Comment:
Why are we making this a part of this interface? If it's to support forking live networks I think this is a premature abstraction and as a tradeoff it bloats this interface. IMO this interface should just return validator sets

Reply:
If we want to support live networks won't we need some timestamp/block height parameter in this interface then?